domain/UserCreateCompany.go

```diff
@@ -57,39 +57,27 @@ func (d *Domain) UserCreateCompany(in *UserCreateCompanyIn) (out UserCreateCompa
 		return
 	}
 
-	if in.TenantCode == `` || !isLetter(in.TenantCode){
-		out.SetError(400, ErrUserCreateCompanyTenantCodeInvalid)
-		return
-	}
-
-	if in.CompanyName == `` {
-		out.SetError(400, ErrUserCreateCompanyCompanyNameInvalid)
-		return
-	}
-
-	if in.HeadTitle == `` {
-		out.SetError(400, ErrUserCreateCompanyHeadTitleInvalid)
-		return
-	}
-
 	user := wcAuth.NewUsersMutator(d.AuthOltp)
 	user.Id = sess.UserId
 	if !user.FindById() {
 		out.SetError(400, ErrUserCreateCompanyUserNotFound)
 		return
 	}
 
+	L.Print(`company:`, in.Company)
+
+	if !isLetter(in.Company.TenantCode) {
+		out.SetError(400, ErrUserCreateCompanyTenantCodeInvalid)
+		return
+	}
+
 	org := wcAuth.NewOrgsMutator(d.AuthOltp)
-	org.SetTenantCode(fmt.Sprintf("%s-%s", in.TenantCode, generate4RandomNumber()))
-	org.SetHeadTitle(in.HeadTitle)
-	org.SetName(in.CompanyName)
+	org.SetName(in.Company.Name)
+	org.SetTenantCode(fmt.Sprintf("%s-%s", in.Company.TenantCode, generate4RandomNumber()))
```

Comment:
this still not idempotent?

this will update the tenant code again and again with new value

Reply:
so to make it idempotent must be something like this:
if tenant code used by other company, return error
if have company in the past, do not generate new tenant code, do not update createdAt